patient-care/webapp/client/shared/widgets.js

```diff
@@ -318,27 +350,44 @@ Template.contactUsButton.helpers({
 Template.listSamplesButton.onCreated(function () {
   let instance = this;
 
-  let showAllDefault = instance.data.length <= 5;
-  instance.showAllSamples = new ReactiveVar(showAllDefault);
+  instance.showMore = new ReactiveVar(false);
+
+  // set the showMore default value whenever the data changes
+  instance.autorun(() => {
+    let { sampleLabels } = Template.currentData();
 
-  instance.hideStudyLabels = new ReactiveVar(false);
+    if (sampleLabels) {
+      instance.showMore.set(sampleLabels.length <= 6);
+    }
+  });
 });
 
 Template.listSamplesButton.helpers({
-  showAllSamples() { return Template.instance().showAllSamples.get(); },
-  hideStudyLabels() { return Template.instance().hideStudyLabels.get(); },
+  showMore() { return Template.instance().showMore.get(); },
+  showStudyLabels() {
+    let { profile } = Meteor.user();
+
+    return profile && profile.showStudyLabels;
+  },
   sampleToShow() {
     let instance = Template.instance();
 
-    let sampleLabels = instance.data;
+    let { sampleLabels } = instance.data;
 
     // remove study labels if necessary
-    if (instance.hideStudyLabels.get()) {
+    let { profile } = Meteor.user();
+    if (!profile || !profile.showStudyLabels) {
       sampleLabels = MedBook.utility.unqualifySampleLabels(sampleLabels);
     }
 
     // return either the whole list or the first couple items
-    if (instance.showAllSamples.get()) {
+    if (instance.showMore.get()) {
+      if (instance.data.sampleLabels.length > 1000) {
+        return sampleLabels
+          .slice(0, 1000)
+          .concat([`... and ${sampleLabels.length - 1000} more samples`]);
```

Comment:
Limiting the list to 1000 is important because it takes a long time to show the list if it's larger than that. If the user wants the whole list they can just download it. There's a nice popup that tells them that they're not getting the whole list. (See the related `.html`.)